controllers/helmchart_controller.go

```diff
@@ -1233,7 +1292,7 @@ func observeChartBuild(obj *sourcev1.HelmChart, build *chart.Build, err error) {
 		}
 
 		switch buildErr.Reason {
-		case chart.ErrChartMetadataPatch, chart.ErrValuesFilesMerge, chart.ErrDependencyBuild, chart.ErrChartPackage:
+		case chart.ErrChartMetadataPatch, chart.ErrValuesFilesMerge, chart.ErrDependencyBuild, chart.ErrChartPackage, chart.ErrChartVerification:
```

Comment:
A few lines above this, we check if the build was successful and SourceVerified=True condition is added.
But when a build wasn't successful, a build error with ChartVerificationError, we should add SourceVerified=False condition. Without this, SourceVerified is only marked to be false when there's a failure in gathering the verification material, and not when verification itself fails.